primitives/src/bazaar.rs

```diff
@@ -14,7 +14,53 @@
 // You should have received a copy of the GNU General Public License
 // along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
 
-use crate::common::IpfsCid;
+use codec::{Decode, Encode};
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_core::RuntimeDebug;
 
-pub type ShopIdentifier = IpfsCid;
-pub type ArticleIdentifier = IpfsCid;
+use crate::common::PalletString;
+use crate::communities::CommunityIdentifier;
+
+#[derive(Encode, Decode, Default, RuntimeDebug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct BusinessIdentifier<AccountId> {
+    pub community_identifier: CommunityIdentifier,
+    pub business_account: AccountId,
+}
+
+impl<AccountId> BusinessIdentifier<AccountId> {
+    pub fn new(cid: CommunityIdentifier, bid: AccountId) -> BusinessIdentifier<AccountId> {
+        BusinessIdentifier {
+            community_identifier: cid,
+            business_account: bid,
+        }
+    }
+}
+
+#[derive(Encode, Decode, Default, RuntimeDebug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct BusinessData {
+    pub url: PalletString,
+    pub last_oid: u32,
+}
+
+impl BusinessData {
+    pub fn new(url: PalletString, last_oid: u32) -> BusinessData {
```

Comment:
shouldn't new BusinessData always be initialized with last_oid = 0?

Reply:
this constructor is mainly used in the tests... they use it to verify, that last_oid does modified, when updates to the business or erroneous calls that should not alter data at all are being called

Reply:
on second thought, this issue might resolve itself when I use the event system to properly do the checks. We will see tomorrow afternoon. :)